File: backend/src/main/java/tom/projects/monkeylog/model/exercise/Exercise.java
package tom.projects.monkeylog.model.exercise;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;
import tom.projects.monkeylog.model.user.UserOwned;
import tom.projects.monkeylog.model.workout.ExerciseGroup;

import java.util.List;
import java.util.UUID;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Exercise implements UserOwned {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;
    private UUID userId;
    private String name;
    @OneToMany(mappedBy = "exercise")
    private List<ExerciseGroup> exerciseGroups;
    @Enumerated(EnumType.STRING)
    private ExerciseCategory exerciseCategory;

}

File: backend/bin/main/tom/com/monkeylog/model/exercise/Exercise.kt
package tom.com.monkeylog.model.exercise

import jakarta.persistence.*
import org.hibernate.annotations.UuidGenerator
import tom.com.monkeylog.model.user.UserOwned
import tom.com.monkeylog.model.workout.ExerciseGroup
import java.util.*

@Entity
class Exercise(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID? = null,
    override var userId: UUID? = null,
    var name: String,
    @OneToMany(mappedBy = "exercise")
    var exerciseGroups: List<ExerciseGroup> = ArrayList(),
    @Enumerated(EnumType.STRING)
    var exerciseCategory: ExerciseCategory
) : UserOwned
